### PR TITLE
[Test] As a logged in user, I can see the survey cards on the dashboard

### PR DESCRIPTION
**Pull request by tyrro**
## What happened 👀
This PR completes unit test suits for https://github.com/tyrro/react-survey/pull/56/files
 
## Insight 📝
 
 This PR `DRY` es the usage of `next/image` into a separate `Image` component
 
## Proof Of Work 📹

all tests are passing!

### CODE REVIEW

File: components/Alert/index.jsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import Image from 'next/image';
+import Icon from '@/components/Icon';
```
Comment (tyrro) [inline components/Alert/index.jsx:1]: DRY-ed usage of `Image` into a `Icon` component!

File: components/Icon/index.test.jsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    render(
+      <div data-test-id={dataTestId}>
+        <Icon {...props} />
+      </div>,
+    );
+
+    const icon = screen.getByTestId(dataTestId);
+
+    expect(icon).toBeVisible();
+    expect(icon).toHaveTextContent(props.src);
```
Comment (malparty) [inline components/Icon/index.test.jsx:23]:
```suggestion
    const image = icon.getByRole('img');

    expect(image).toHaveAttribute('src', props.src);
```
Should we be more specific to ensure the src is not contained in an error message? (e.g. if the image component would raise an error "the image path ${props.src} is invalid", this test would still pass.

Though my suggestion has a limitation: we assume things about the image component while we do not have direct control over it (as it comes from `next/image`)

File: adapters/Survey/index.test.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import baseAdapter from 'adapters/Base';
+import surveyAdapter from '.';
+
+describe('surveyAdapter', () => {
+  describe('fetchSurveys', () => {
+    describe('given the current page and an authorization header', () => {
```
Comment (malparty) [inline adapters/Survey/index.test.js:6]:
```suggestion
    describe('given the current page is 1 and an authorization header', () => {
```

or 

```suggestion
    describe('given a current page param and an authorization header', () => {
```

File: components/Icon/index.jsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+import PropTypes from 'prop-types';
+
+const Icon = ({ src, alt, ...attributes }) => <Image src={src} alt={alt} {...attributes} />;
+
+Icon.propTypes = {
+  src: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
+  alt: PropTypes.string.isRequired,
+};
+
+export default Icon;
```
Comment (malparty) [inline components/Icon/index.jsx:11]: As its usage is not restricted to `icons`, should we name it "Image" instead? 

You might need to play with aliases, something like:

```
import NextImage from 'next/image';
```

File: components/Icon/index.test.jsx (hunk 2 above)
Comment (tyrro) [reply components/Icon/index.test.jsx]: Re-wrote `next/image` mock to test image by `src` attribute in 30b5d8340cd5b12952568fa9b4f326451f3c58ed

File: adapters/Survey/index.test.js (hunk 3 above)
Comment (tyrro) [reply adapters/Survey/index.test.js]: done in e9f3815051be75cb8f3734494b6d05c9fc389acf

File: components/Icon/index.jsx (hunk 4 above)
Comment (tyrro) [reply components/Icon/index.jsx]: sure, done in e9f3815051be75cb8f3734494b6d05c9fc389acf

**Review by malparty — APPROVED** [review]
LGTM :) 
Please just add a small insight about why you needed the custom Image component and a small indication of Proof of Work (all in the PR description). Thanks! 🙌

**Comment by tyrro** [comment]
> LGTM :) Please just add a small insight about why you needed the custom Image component and a small indication of Proof of Work (all in the PR description). Thanks! 🙌
>

done, Thank you!

File: adapters/Survey/index.test.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import baseAdapter from 'adapters/Base';
+import surveyAdapter from '.';
+
+describe('surveyAdapter', () => {
+  describe('fetchSurveys', () => {
+    describe('given a current page and an authorization header', () => {
```
Comment (carryall) [inline adapters/Survey/index.test.js:6]: Maybe adding this to make it a bit more clear 💡 

```suggestion
    describe('given a current page number and an authorization header', () => {
```

**Review by carryall — APPROVED** [review]
LGTM with a small comment and do you plan to have another PR to add UI tests? 🤔

File: adapters/Survey/index.test.js (hunk 5 above)
Comment (tyrro) [reply adapters/Survey/index.test.js]: done in ab05c62c262d7d61b22889922d4d4962fe36db17!

**Comment by tyrro** [comment]
> LGTM with a small comment and do you plan to have another PR to add UI tests? 🤔
>
Yes, I plan to work on UI tests in a separate PR! Thank you!

**Pull request merged**